k8s/api.yaml

```diff
@@ -0,0 +1,100 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: api
+      app.kubernetes.io/instance: api
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: api
+        app.kubernetes.io/instance: api
+    spec:
+      containers:
+        - name: api
+          image: iteam1337/brevduvor-api:latest
+          imagePullPolicy: Always
+          ports:
+            - name: http
+              containerPort: 4000
+              protocol: TCP
+          resources: {}
+          envFrom:
+            - secretRef:
+                name: postgres-password
+          env:
+            - name: OSRM_HOST
+              value: http://osrm.default.svc.cluster.local
+            - name: DRONE_URL
+              value: https://drone.brevduvor.iteamdev.se
+            - name: WEBHOOK_URL
+              value: https://api.brevduvor.iteamdev.se/status
+            - name: ELASTIC_HOST
+              value: http://10.244.1.140:9200
```

Comment:
Why is the Elastic host an IP instead of a hostname?

Also I see you use the full name above http://osrm.default.svc.cluster.local so you can use the same format for elastic also:
`<service>.<namespace>.svc.cluster.local`

Reply:
Not super sure about this, and it's definately something i want to fix. But right now, elasticsearch.yaml/service is of type NodePort. From what I understand, deploying the service as a ClusterIP lets you connect to it with the hostname. But I couldn't  get that working.